cpp/src/Slice/Parser.h

```diff
@@ -363,10 +363,11 @@ namespace Slice
     {
     public:
         [[nodiscard]] ContainerPtr container() const;
-        [[nodiscard]] std::string name() const;
-        [[nodiscard]] std::string scoped() const;
-        [[nodiscard]] std::string scope() const;
-        [[nodiscard]] std::string flattenedScope() const;
```

Comment:
`flattenedScope` was only used by `slice2cpp`, so I moved the logic into that compiler.